tmt/base.py

```diff
@@ -2849,6 +2849,7 @@ def __init__(self,
                  path: Optional[Path] = None,
                  tree: Optional[fmf.Tree] = None,
                  fmf_context: Optional[FmfContext] = None,
+                 additional_rules: Optional[list[Any]] = None,
```

Comment:
The type should probably be `Optional[list[_RawAdjustRule]]`, it's what we feed `adjust` (https://github.com/teemtee/tmt/blob/main/tmt/base.py#L689). Or, if not and `when` & `description` are not to appear in `adjust-test`, then we should have a new type for these rules, mostly because `Any` is really, really hard to track through code, while `_RawAdjustTestsRule` is way easier to recognize.

Reply:
attempt in 3f12f70e